addons/mail/models/template_reset_mixin.py

```diff
@@ -0,0 +1,140 @@
+
+        if trans_to_remove:
+            self.env['ir.translation'].search([
+                ('name', 'like', f'{self._name},%'),
+                ('res_id', '=', self.id),
+                ('lang', 'in', trans_to_remove),
+                ('module', 'in', [module_name, False]),
+            ]).unlink()
+
+    def _process_translation_data(self, trans_file, lang, xml_ids):
+        """Populates the ir_translation table.
+
+        :param trans_file: path to a translation file to open, e.g. {addon_path}/mail/i18n/es.po
+        :param lang: language code of the translations contained in `trans_file`
+                     language must be present and activated in the database
+
+        :return Boolean: True if translation data is available and processed, False otherwise
+        """
+        with file_open(trans_file, mode='rb', filter_ext=(".po",)) as fileobj:
+            translation_processed = False
+            reader = TranslationFileReader(fileobj)
+            # Process a single PO entry
+            for row in reader:
+                if row.get('imd_name') in xml_ids:
+                    translation_processed = True
+                    # copy Translation from Source to Destination object
+                    self.env['ir.translation']._set_ids(
+                        row['name'], 'model', lang, self._ids, row.get('value', ''), row['src'],
+                    )
+            return translation_processed
+
+    def reset_template(self):
+        """Resets the Template with values given in source file. We ignore the case of
+        template being overridden in another modules because it is extremely less likely
+        to happen. This method also tries to reset the translation terms for the current
+        user lang (all langs are not supported due to costly file operation). """
+        expr = "//*[local-name() = $tag and (@id = $xml_id or @id = $external_id)]"
+        for template in self.filtered('template_fs'):
+            external_id = template.get_external_id().get(template.id)
+            module, xml_id = external_id.split('.')
+            fullpath = get_resource_path(*template.template_fs.split('/'))
+            if fullpath:
+                doc = etree.parse(fullpath)
+                for rec in doc.xpath(expr, tag='record', xml_id=xml_id, external_id=external_id):
+                    # We don't have a way to pass context while loading record from a file, so we use this hack
+                    # to pass the context key that is needed to reset the fields not available in data file
+                    rec.set('context', json.dumps({'reset_template': 'True'}))
+                    obj = xml_import(template.env.cr, module, {}, mode='init', xml_filename=fullpath)
+                    obj._tag_record(rec)
+                    template._override_translation_term(module, [xml_id, external_id])
+            else:
+                raise UserWarning(_("Could not reset the template."))
```

Comment:
Maybe "We could not reset templates %(template_name)s as it is not linked to an Odoo source file", something like that ?